### DISCUSSION TITLE
Should I avoid updating a signal without actual changes?

### DISCUSSION THREAD

**Discussion by kaivol** [lapce/floem#449] [General]
Before calling `RwSignal::set` I could always check that the new value is actually different from the current one. 
Does such a check make sense performance-wise (at least in the case that actual changes are rather rare) or is it always redundant?

Thanks!

**Comment by dzhou121** [comment]
You can use Memo
https://docs.rs/floem_reactive/0.1.0/floem_reactive/struct.Memo.html

**Reply by kaivol** [reply]
Thanks for the tip!

I'll take that as a general yes, right? 

Also, does it make a difference whether I do the check myself or use `Memo`?

**Reply by jrmoulton** [reply]
I would actually say generally no. You would only want to use a memo if there are effects that would run in response to setting the signal that would be expensive and you wouldn't want to rerun more than necessary. I would say that this is much more rare than common though.

**Reply by kaivol** [reply]
So the use case that I have in mind is a theming system:
I have a big struct `Colors` containing a large number of colors which is accessed in the top-level `.style()` call as well as in multiple places across the application. It can be changed manually by the user or automatically as a result of OS events (e.g. dark/light mode). 
Would that qualify as _"expensive effects"_, or does that only refer to effects such as file access or other I/O?

Or should I perhaps rather split this large struct into multiple individual `RwSignal`s like this:
````rust
struct Colors {
    background: RwSignal<Color>, 
    foreground: RwSignal<Color>, 
    ...
}
````
